lib/explorer/data_frame.ex

```diff
@@ -1122,8 +1141,64 @@ defmodule Explorer.DataFrame do
     df |> callback.() |> then(&mask(df, &1))
   end
 
-  @doc false
-  def filter_with(df, fun) when is_function(fun) do
+  @doc """
+  Picks rows based on a callback function. 
+
+  This function is efficient because it uses a representation of the
+  series without pulling them. The only restriction is that
+  you need to use one of the following functions to perform a comparison:
```

Comment:
I don't think it is a matter of the function but rather a matter of returning booleans?